### PR TITLE
feat(chatgpt): update to streaming and new ChatGPT API

### PR DESCRIPTION
**Pull request by dominiksipowicz**
### Description

- reused [OpenAIStream.ts](https://github.com/vercel/examples/compare/chat-gpt-update?expand=1#diff-a5e6c5d666557cf9baf9310bd3e97f11a41636cdb3112b1a16ebe83b0b9b095c) from our other examples
- removed a lot of code and made it simpler.
- removed `AI_PROMPT` to simplify implementation

### How to test?

- test locally on the branch
- `cd solutions/ai-chatgpt`
- `cp .env.example .env.local`
- update OPENAI_API_KEY with your [OpenAI](https://beta.openai.com/account/api-keys) secret key.
- `npm run dev`

### Type of Change

- [x] Example updates (Bug fixes, new features, etc.)

### New Example Checklist

- [x] 📱 Is it responsive? Are mobile and tablets considered?

### CODE REVIEW

File: solutions/ai-chatgpt/components/Chat.tsx [hunk 1]
```diff
@@ -58,42 +58,68 @@ export function Chat() {
...
+  const newMessages = [
+    ...messages,
+    { role: "user", content: message } as ChatGPTMessage,
+  ];
+  setMessages(newMessages);
+  const last10messages = newMessages.slice(-10); // remember last 10 messages
+
+  const response = await fetch("/api/chat", {
+    method: "POST",
+    headers: {
```
Comment (lfades) [inline solutions/ai-chatgpt/components/Chat.tsx:74]: The indentation here is wrong, the function is still part of `Chat`. Can you run prettier over the files here 🙏

File: solutions/ai-chatgpt/utils/OpenAIStream.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            // maybe parse error
+            controller.error(e);
+          }
+        }
+      }
+
+      // stream response (SSE) from OpenAI may be fragmented into multiple chunks
+      // this ensures we properly read chunks and invoke an event for each SSE event stream
+      const parser = createParser(onParse);
+      // https://web.dev/streams/#asynchronous-iteration
```
Comment (lfades) [inline solutions/ai-chatgpt/utils/OpenAIStream.ts:81]:
```suggestion
```

**Comment by Nutlope** [comment]
LGTM!

File: solutions/ai-chatgpt/components/Chat.tsx (hunk 1 above)
Comment (dominiksipowicz) [reply solutions/ai-chatgpt/components/Chat.tsx]: Good call. I really don't know why it broke prettier. I had to reinstall it and it worked.

**Review by lfades — APPROVED** [review]

**Pull request merged**